src/server/http/__tests__/xsrf.js

```diff
@@ -29,10 +30,23 @@ describe('xsrf request filter', function () {
 
     await kbnServer.ready();
 
-    const routeMethods = nonDestructiveMethods.filter(method => method !== 'HEAD').concat(destructiveMethods);
     kbnServer.server.route({
       path: '/xsrf/test/route',
-      method: routeMethods,
+      method: nonDestructiveMethods.filter(method => method !== 'HEAD'),
+      handler: function (req, reply) {
+        reply(null, 'ok');
+      }
+    });
+
+    kbnServer.server.route({
+      path: '/xsrf/test/route',
+      method: destructiveMethods,
+      config: {
+        // Disable payload parsing to make HapiJS server accept any content-type header.
```

Comment:
> What happens in our "production setup" vs "test setup" here? If we special-case the config here, but it isn't specified like this in our actual routes, will it work as expected?

In  "production setup" raw request payload is parsed and converted to the object. But for this particular case (testing of xsrf "subsystem") I think there shouldn't be any difference as we don't care about the payload at all.

> Maybe we should add 1 or 2 integration tests that checks for the 400 response at least? See test/api_integration/apis in https://github.com/elastic/kibana/pull/11114/files for some examples.

Well I think it may be reasonable to have two different test routes in the test. The first one similar to the majority of our routes, as it was before and the second one for route with disabled payload parsing - like we have for `_msearch` for example. With this I'll test `x-ndjson` only against that second route that is close to real setup.

What do you think?

Regarding 400 response, what xsrf-related use cases do you have in mind for that? We have few `rejects requests ...` related tests already...